### ISSUE TITLE
ArticleFeed misses Filterattributes that are linked  to Child-Articles

### ISSUE THREAD

**Issue by m-schiewi** [FACT-Finder-Web-Components/oxid-eshop-module#98]
Imagine an article "MyFavouriteCar" available in two variants for motor power having 100 HP (= horse power) and 150 HP respectivly.  Database would contain three records as follows: parent_mycar, child_100hp, child_150hp).OXVARNAME and OXVARSELECT are set accordingly for the 100HP and 150HP

Imagine attribute "Fuel Consumption" exists in the shop. Value assignments would be like:
- child_100hp: "6 l/100km"
- child_150hp: "10 l/100km"
(it's obvious, that the stronger powered car has a higher fuel consumption)

You could also imagine an attribute called "Energy label"
- child_100hp: "A++"
- child_150hp: "A+"

Even if these attributes are enlisted as "filter attributes" in module settings, its values are not contained in the article feed.

Root cause can be found in FilterAttributes.php:29-33
in short: if $article is not parent, only the corresponding variant value is collected. Attributes assigned to the child are ignored.

General remark:
attributes value atttached to child articles usually show a direct functional dependence to the corresponding variant value. Here: fuelconsumption = f(horsepower) or energy_label = f(horse_power).  Customer must have the possibility to filter for these values.

**Comment by a-laurowski** [comment]
hi @m-schiewi.
Thanks for report. We are aware that in most of our SDK's the FilterAttribute value is mostly taken from the "parent level" (except the configurable attributes). That was an assumption when we were working on the export, but right now, we are receiving more and more requests about changing that. We already have "make FilterAttributes takes values from the variants" in our roadmap. Due to overall amount of work i can't provide more specific date than March 2022.

For the time being i recommend to extend our export with the custom field provider. With that approach the attribute which values has to be exported from variants must be a separate column.


1. Create a field provider that implements [FieldInterface](https://github.com/FACT-Finder-Web-Components/oxid-eshop-module/blob/master/src/Export/Field/FieldInterface.php) - in method `getValue` you will receive both: current variant and its parent
2.  Override this method from [https://github.com/FACT-Finder-Web-Components/oxid-eshop-module/blob/31ad9ac0aa487e244ae68dc18bc727ff5ed68d30/src/Export/ArticleFeed.php#L49](ArticleFeed) and register created field provider.

After doing that, a new column should be added to the feed with values taken from the newly created field provider.

**Comment by a-laurowski** [comment]
hi @m-schiewi  The new release `v4.2.0` that contains a proper change to `FilterAttributes` export is out
https://github.com/FACT-Finder-Web-Components/oxid-eshop-module/releases/tag/v4.2.0
